### PR TITLE
[Community Pipeline] MagicMix

### PR DESCRIPTION
**Pull request by daspartho**
Community pipeline based on [my implementation](https://github.com/daspartho/MagicMix) of the [MagicMix: Semantic Mixing with Diffusion Models](https://arxiv.org/abs/2210.16056) paper. 

This Diffusion Pipeline allows for the semantic mixing of an image and a text prompt to create a new concept while preserving the spatial layout and geometry of the subject in the image.

Here are some examples I reproduced from the paper using my implementation-

##### Input Image:

telephone

##### Output Image:

telephone-bed

##### Input Image:

sign

##### Output Image:

sign-family

##### Input Image:

sushi

##### Output Image:

sushi-ice-cream

##### Input Image:

pineapple

##### Output Image:

pineapple-cake

@patrickvonplaten could you please take a look at it, looking forward to any comments! 
Thanks :)

Reference: #841

### CODE REVIEW

**Comment by aengusng8** [comment]
Cool! But why did you put it in the **community pipeline** instead of the **internal pipeline**?

File: examples/community/README.md [hunk 1]
```diff
@@ -815,6 +816,50 @@ plt.title('Stable Diffusion v1.4')
...
+Here is an example usage-
+
 ```python
+from diffusers import DiffusionPipeline, DDIMScheduler
+from PIL import Image
+
+pipe = DiffusionPipeline.from_pretrained(
+    "CompVis/stable-diffusion-v1-4",
+    custom_pipeline="magic_mix",
+    scheduler = DDIMScheduler(beta_start=0.00085, beta_end=0.012, beta_schedule="scaled_linear", clip_sample=False, set_alpha_to_one=False),
```
Comment (patil-suraj) [inline examples/community/README.md:840]: The scheduler can be loaded using 

```python
DDIMScheduler.from_pretrained("CompVis/stable-diffusion-v1-4", subfolder="scheduler")
```

File: examples/community/README.md [hunk 2]
```diff
@@ -815,6 +816,50 @@ plt.title('Stable Diffusion v1.4')
...
+
+Here is an example usage-
+
 ```python
+from diffusers import DiffusionPipeline, DDIMScheduler
+from PIL import Image
+
+pipe = DiffusionPipeline.from_pretrained(
+    "CompVis/stable-diffusion-v1-4",
+    custom_pipeline="magic_mix",
```
Comment (patil-suraj) [inline examples/community/README.md:839]: Also, maybe load the pipeline in `fp16`, by passing the `torch_dtype` argument, to make inference faster.

File: examples/community/magic_mix.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+        return torch.cat([uncond_embedding, text_embedding])
+
+    def __call__(
+        self,
+        img: Image.Image,
+        prompt: str,
+        kmin: float = 0.3,
+        kmax: float = 0.6,
+        v: float = 0.5,
```
Comment (patil-suraj) [inline examples/community/magic_mix.py:79]: could we use a more descriptive name for this argument, one letter variables aren't informative

**Review by patil-suraj — APPROVED** [review]
Very cool @daspartho ! The PR looks good, just left some nits

**Comment by daspartho** [comment]
@patil-suraj made some changes :)

**Comment by daspartho** [comment]
The `v` parameter is the interpolation constant used in the layout generation process, so I settled for `mix_factor` as the new parameter name.
It is clear and more descriptive, and it helps to convey the purpose of the parameter in the context of the code.

wyt @patil-suraj

**Review by patil-suraj — APPROVED** [review]
Thanks a lot! `mix_factor` sounds good to me.
